icu4c/source/test/intltest/numbertest_permutation.cpp

```diff
@@ -0,0 +1,189 @@
+    { "zh-TW", u"zh-TW" },
+    { "bn-BD", u"bn-BD" },
+  };
+
+  // Convert kSkeletonParts to a more convenient data structure
+  auto skeletonParts = std::vector<std::vector<const char16_t *>>();
+  auto currentSection = std::vector<const char16_t *>();
+  for (int32_t i = 0; i < UPRV_LENGTHOF(kSkeletonParts); i++) {
+    const char16_t *skeletonPart = kSkeletonParts[i];
+    if (skeletonPart == 0) {
+      skeletonParts.push_back(currentSection);
+      currentSection.clear();
+    } else {
+      currentSection.push_back(skeletonPart);
+    }
+  }
+
+  // Build up the golden data string as we evaluate all permutations
+  std::vector<UnicodeString> results;
+
+  // Take combinations of 3 orthogonal options
+  for (size_t i = 0; i < skeletonParts.size() - 2; i++) {
+    const auto& skeletons1 = skeletonParts[i];
+    for (size_t j = i + 1; j < skeletonParts.size() - 1; j++) {
+      const auto& skeletons2 = skeletonParts[j];
+      for (size_t k = j + 1; k < skeletonParts.size(); k++) {
+        const auto& skeletons3 = skeletonParts[k];
+
+        // Evaluate all combinations of skeletons for these options
+        for (const auto& skel1 : skeletons1) {
+          for (const auto& skel2 : skeletons2) {
+            for (const auto& skel3 : skeletons3) {
+
+              // Compute the skeleton
+              UnicodeString skeleton;
+              skeleton.append(skel1)
+                  .append(u' ')
+                  .append(skel2)
+                  .append(u' ')
+                  .append(skel3);
+              results.push_back(skeleton);
+
+              // Check several locales and several numbers in each locale
+              for (const auto& locData : localesToTest) {
+                auto lnf = NumberFormatter::forSkeleton(skeleton, status)
+                    .locale(locData.locale);
+                results.push_back(UnicodeString(u"  ").append(locData.ustring));
+                for (const auto& input : kNumbersToTest) {
+                  results.push_back(UnicodeString(u"    ").append(
+                      lnf.formatDouble(input, status).toTempString(status)));
+                }
+              }
+
+              results.push_back(UnicodeString());
+            }
+          }
+        }
+      }
+    }
+  }
+
+  CharString goldenFilePath(getSourceTestData(status), status);
+  goldenFilePath.appendPathPart("numberpermutationtest.txt", status);
+
+  // Compare it to the golden file
+  const char *codePage = "UTF-8";
+  LocalUCHARBUFPointer f(ucbuf_open(goldenFilePath.data(), &codePage, TRUE, FALSE, status));
+  if (!assertSuccess("Can't open data file", status)) {
+    return;
+  }
+
+  int32_t lineNumber = 1;
+  int32_t lineLength;
+  for (const auto& actualLine : results) {
+    const UChar* lineBuf = ucbuf_readline(f.getAlias(), &lineLength, status);
+    if (lineBuf == nullptr) {
+      errln("More lines generated than are in the data file!");
+      break;
+    }
+    UnicodeString expectedLine(lineBuf, lineLength - 1);
```

Comment:
How about skipping comment lines, and empty lines following comments or other empties?
For example, I think we should prepend a copyright header.

Reply:
The design of this test is to treat the file as a golden file, not intended to be edited by humans except when using the `-G` flag to intltest.

I added a comment block.